mls/nostr-mls/src/messages.rs

```diff
@@ -480,6 +480,10 @@ where
         // Save exporter secret for the new epoch
         self.exporter_secret(mls_group.group_id())?;
 
+        // Sync the stored group metadata with the updated MLS group state
+        // This ensures any group context extension changes are reflected in storage
+        self.sync_group_metadata_from_mls(mls_group.group_id())?;
+
```

Comment:
Should we look for a way to 'undo' the previous things if this sync fails (basically see if we can make it transactional)? I'm not sure what happens in that case